src/flash_devices.h

```diff
@@ -525,4 +525,17 @@ typedef struct {
     .single_status_byte = false, .is_fram = false,                             \
   }
 
+// Settings for the Puya Semiconductor P25Q16H 2MiB QSPI flash.
+// Datasheet: https://www.puyasemi.com/uploadfiles/2021/12/202112201130233023.pdf
+#define P25Q16H                                                                \
```

Comment:
run `clang-format -i src/*` to get it formatted correctly. If you are not familliar with clang, you can replace it with text suggested by ci